src/flint/gfp_fmpz_mat.jl

```diff
@@ -281,14 +267,6 @@ function (a::FpMatrixSpace)()
   return z
 end
 
-function (a::FpMatrixSpace)(b::IntegerUnion)
-   M = a()  # zero
-   for i in 1:min(nrows(a), ncols(a))
-      M[i, i] = base_ring(a)(b)
-   end
-   return M
-end
-
 function (a::FpMatrixSpace)(b::FpFieldElem)
```

Comment:
I would expect this one to work generically

Reply:
In this case that might actually be true -- but still, I'd really prefer to move forward now, and if you or someone else would like to remove further methods in a follow-up, be my guest...